subset/alliedSwitch/src/main/java/alliedswitch/AlliedInterrogator.java

```diff
@@ -0,0 +1,462 @@
+  String[] show_interface_data = new String[21];
+
+  String[] show_interface_port_expected = {"port1", "\n", "Time since last state change: ", "\n"};
+  int[] show_interface_port_pointers = new int[4];
+
+  String[] show_platform_expected = {
+    "port1",
+    "\n",
+    "enabled:",
+    "\n",
+    "loopback:",
+    "\n",
+    "link:",
+    "\n",
+    "speed:",
+    "m",
+    "max speed:",
+    "\n",
+    "duplex:",
+    "\n",
+    "linkscan:",
+    "\n",
+    "autonegotiate:",
+    "\n",
+    "master:",
+    "\n",
+    "tx pause:",
+    "r",
+    "rx pause:",
+    "\n",
+    "untagged vlan:",
+    "\n",
+    "vlan filter:",
+    "\n",
+    "stp state:",
+    "\n",
+    "learn:",
+    "\n",
+    "discard:",
+    "\n",
+    "jam:",
+    "\n",
+    "max frame size:",
+    "\n",
+    "MC Disable SA:",
+    "\n",
+    "MC Disable TTL:",
+    "\n",
+    "MC egress untag:",
+    "\n",
+    "MC egress vid:",
+    "\n",
+    "MC TTL threshold:",
+    "\n"
+  };
+  int[] show_platform_pointers = new int[48];
+  String[] show_platform_data = new String[24];
+
+  String[] show_platform_port_expected = {"port1", "\n", "MC TTL threshold:", "\n"};
+  int[] show_platform_port_pointers = new int[4];
+
+  int number_switch_ports = 1; // 48
+
+  int requestFlag = 0;
+
+  public int getRequestFlag() {
+    return requestFlag - 1;
+  }
+
+  int count = 0;
+
+  String device_hostname;
+
+  String login_report = "";
+
+  int continue_flag = 1;
+
+  boolean debug = true;
+
+  boolean extendedTests = false;
+
+  public AlliedInterrogator(String remoteIpAddress, int interfacePort) {
+    this.remoteIpAddress = remoteIpAddress;
+    this.interfacePort = interfacePort;
+    command[interfacePos] = command[interfacePos] + interfacePort;
+    command[platformPos] = command[platformPos] + interfacePort;
+  }
+
+  @Override
+  public void run() {
+    System.out.println("Interrogator new connection...");
+    telnetClientSocket = new AlliedTelnetClientSocket(remoteIpAddress, remotePort, this, debug);
+    telnetClientSocketThread = new Thread(telnetClientSocket);
+    telnetClientSocketThread.start();
+  }
+
+  public void receiveData(String data) {
+    if (debug) {
+      System.out.println(
+          java.time.LocalTime.now() + "receiveDataLen:" + data.length() + "receiveData:" + data);
+    }
+
+    Runnable runnable =
+        () -> {
+          parseData(data);
+        };
+    Thread parseThread = new Thread(runnable);
+
+    parseThread.start();
+  }
+
+  private void parse_packet(String raw_data, String[] show_expected, int[] show_pointers) {
+    int start = 0;
+    for (int port = 0; port < number_switch_ports; port++) {
+      for (int x = 0; x < show_expected.length; x++) {
+        start = recursive_data(raw_data, show_expected[x], start);
+        show_pointers[x] = start;
+      }
+
+      int chunk_s = show_pointers[0];
+      int chunk_e = show_pointers[3];
+
+      if (chunk_e == -1) chunk_e = raw_data.length();
```

Comment:
Surprised the style check didn't catch this -- but don't have if statements without {} -- it's more consistent overall to always have {. so, it should be:
```

 if (chunk_e == -1) {
  chunk_e = raw_data.length();
}

```